version: '4.29'

services:
  mysql_server:
    build:
      context: ./mysqldb
      dockerfile: Dockerfile
    restart: always
    env_file:
      - .env
    environment:
      - MYSQL_ROOT_PASSWORD=${ROOT_PASSWORD}
      - MYSQL_DATABASE=${DATABASE} 
      - MYSQL_USER=${USER}
      - MYSQL_PASSWORD=${PASSWORD}
    # ports:
    #   - "3306:3306"
    healthcheck:
      test: ["CMD", "mysql", "-h", "localhost", "-u", "root", "root_password", "-e", "SELECT 1"]
      timeout: 20s
      retries: 10
    networks:
      - app_network
        # ipv4_address: 172.20.0.2

  recipesystem:
    build:
      context: ./django
      dockerfile: Dockerfile
    # command: python manage.py runserver 0.0.0.0:8000
    command: sh -c "python manage.py migrate --noinput && python manage.py runserver 0.0.0.0:8000"
    restart: always
    volumes:
      - ./django:/app
      - ./AIML:/app/AIML
    ports:
      - "8000:8000"
    depends_on:
      mysql_server:
        condition: service_healthy
    networks:
      - app_network

networks:
  app_network:
    driver: bridge
    ipam:
      config:
        - subnet: 172.20.0.0/16